modules/@apostrophecms/schema/index.js

```diff
@@ -1466,7 +1466,6 @@ module.exports = {
           let choices = [];
           if (
             !field ||
-            field.type !== 'select' ||
```

Comment:
This could impact custom field types that might use `choices` differently. I think we'd better keep enumerating the types for now, or the field type definition could have a flag to active this behavior, which you would set for the three  standard types that should have it. You'd check it here via `self.fieldTypes[field.type].flag`. I like that idea.